src/NServiceBus.Core/Routing/IRouteSource.cs

```diff
@@ -0,0 +1,11 @@
+namespace NServiceBus
+{
+    using System;
+    using Routing;
+
+    interface IRouteSource
+    {
+        void GenerateRoutes(Action<RouteTableEntry> registerRouteCallback);
+        RouteSourcePriority Priority { get; }
```

Comment:
would it make sense to move the priority to the `RouteTableEntry` class? This would allow sources to contain routes with different priorities. But since we're not having a specific use case for that, I'm good to keep it this way.